### ISSUE TITLE
Trying to use with dnxcore50 causes lots of errors

### ISSUE THREAD

**Issue by david-driscoll** [NuGet/Home#958]
I'm attempting to add the v3 library from the build feed, and it's giving all sorts of wonderful errors.  I looked at the package it looks correct, so I don't know exactly what is going on at the moment.

I was mainly wondering if it's a known issue or if there was a work around I should be doing instead.
![image](https://cloud.githubusercontent.com/assets/1269157/8727504/132d1012-2bae-11e5-8ddf-ab3408375036.png)

**Comment by yishaigalatzer** [comment]
@david-driscoll can you be more specific about what package and where did you try to use it? Version of visual studio and version of nuget etc.

**Comment by david-driscoll** [comment]
I'm trying to add nuget package search for Omnisharp-roslyn, so editors can
offer project.json intellisense.

I was using both omnisharp-atom _and_ Visual Studio 2015 RC to test.  I
wanted to make sure it wasn't strictly an omnisharp issue.

The packages that I'm using come from (
https://www.myget.org/gallery/nugetbuild ).

At first I thought it was just an issue adding `NuGet.Protocol.Core.v3` but
I also ended up getting the same issue with `NuGet.Configuration` and
`NuGet.Protocol.Core.Types`.

Now it's possible it's a runtime issue, Omnisharp is currently built
against dnx-beta4, and I'm not sure what version the packages that are
published are built against.  Anytime I tried to add the nuget packages as
a dependency for `dnxcore50` I would encounter > 5000 diagnostics for the
solution, all inside the dnxcore project.  Also I realize that
`NuGet.Protocol.Core.v2` is not supported for dnxcore50, so I wasn't trying
to reference that package.

The code is available at:
https://github.com/OmniSharp/omnisharp-roslyn/tree/nuget-packages

I currently have everything ifdef'd so that it will only build against
`dnx451`.  I can build a proper repro branch if it will help.
